### PR TITLE
Enable linting in flow/, and fml/

### PR DESCRIPTION
**Pull request by zanderso**
## Description

This PR removes `// FLUTTER_NOLINT` from source files under `flow/` and `fml/`.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: flow/matrix_decomposition.cc [hunk 1]
```diff
@@ -1,7 +1,6 @@
 // Copyright 2013 The Flutter Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-// FLUTTER_NOLINT
+// found in the LICENSE file.s
```
Comment (gaaclarke) [inline flow/matrix_decomposition.cc:3]: There is an accidental `s` in this line.

File: flow/matrix_decomposition.cc [hunk 2]
```diff
@@ -71,14 +70,14 @@ MatrixDecomposition::MatrixDecomposition(SkM44 matrix) : valid_(false) {
 
   scale_.x = row[0].length();
 
-  SkV3Normalize(row[0]);
+  SkV3Normalize(&row[0]);
```
Comment (gaaclarke) [inline flow/matrix_decomposition.cc:73]: This is so much better.

File: flow/raster_cache.cc [hunk 3]
```diff
@@ -160,10 +159,11 @@ std::unique_ptr<RasterCacheResult> RasterCache::RasterizeLayer(
                                            canvas_size.height());
         internal_nodes_canvas.addCanvas(canvas);
         Layer::PaintContext paintContext = {
-            (SkCanvas*)&internal_nodes_canvas,  // internal_nodes_canvas
-            canvas,                             // leaf_nodes_canvas
-            context->gr_context,                // gr_context
-            nullptr,                            // view_embedder
+            static_cast<SkCanvas*>(
+                &internal_nodes_canvas),  // internal_nodes_canvas
```
Comment (gaaclarke) [inline flow/raster_cache.cc:163]: fyi if you change these argument names to comments like:  `/*internal_nodes_canvas=*/` the linter should make sure they are correct now.

File: flow/raster_cache.h [hunk 4]
```diff
@@ -22,7 +22,7 @@ class RasterCacheResult {
 
   virtual ~RasterCacheResult() = default;
 
-  virtual void draw(SkCanvas& canvas, const SkPaint* paint = nullptr) const;
+  virtual void draw(SkCanvas& canvas, const SkPaint* paint) const;
```
Comment (gaaclarke) [inline flow/raster_cache.h:25]: You can also overload draw with a non-virtual method that just takes in a "canvas" if you want the user code to look the same.

File: flow/matrix_decomposition.cc (hunk 1 above)
Comment (zanderso) [reply flow/matrix_decomposition.cc]: Removed

File: flow/raster_cache.cc (hunk 3 above)
Comment (zanderso) [reply flow/raster_cache.cc]: Done

File: flow/raster_cache.h (hunk 4 above)
Comment (zanderso) [reply flow/raster_cache.h:25]: There were only a couple of call sites, so I left it as is.

**Pull request merged**